libs/@local/hash-graph-types/rust/src/knowledge/entity/provenance.rs

```diff
@@ -7,9 +7,167 @@ use bytes::BytesMut;
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct ArtifactLocation {
+    /// A string containing a valid relative or absolute URI.
+    pub uri: Url,
+
+    /// Encapsulates a message intended to be read by the end user.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub enum SourceType {
+    Webpage,
+    Document,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct SourceProvenance {
+    #[serde(rename = "type")]
+    pub ty: SourceType,
+    pub artifact: Artifact,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub authors: Vec<String>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub actor_type: Option<ActorType>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub first_published: Option<OffsetDateTime>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub last_updated: Option<OffsetDateTime>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub loaded_at: Option<OffsetDateTime>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub analyzed_at: Option<OffsetDateTime>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub enum OriginType {
+    WebApp,
+    MobileApp,
+    BrowserExtension,
+    Api,
+    Flows,
+    Migration,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct OriginProvenance {
+    #[serde(rename = "type")]
+    pub ty: OriginType,
```

Comment:
```suggestion
    pub type: OriginType,
```

Reply:
`type` is not possible as it's a reserved keyword. It's (de-)serialized from/to `type`. This is common in Rust.

Reply:
Ahh, sorry, didn't realise